code/modules/hallucination/mother.dm

```diff
@@ -75,7 +75,19 @@
 	image_state = ""
 
 /obj/effect/client_image_holder/hallucination/your_mother/Initialize(mapload, list/mobs_which_see_us, datum/hallucination/parent)
-	. = ..()
-	var/mob/living/carbon/hallucinator = parent.hallucinator
-	image_icon = getFlatIcon(get_dynamic_human_appearance(/datum/outfit/yourmother, hallucinator.dna.species.type))
-	regenerate_image()
+	var/mob/living/hallucinator = parent.hallucinator
+	if (ishuman(hallucinator))
+		var/mob/living/carbon/dna_haver = hallucinator
+		image_icon = getFlatIcon(get_dynamic_human_appearance(/datum/outfit/yourmother, dna_haver.dna.species.type))
+		return ..()
+
+	if (istype(hallucinator, /mob/living/basic/pet/dog/corgi/ian))
```

Comment:
this has made me ponder whether the bartender has pun pun as a pet or as an adopted child